lib/src/unsafe/io.dart

```diff
@@ -22,7 +22,7 @@ Future unsafeIOInterpreter(IOOp io) {
     return new Future.error(io.failure);
 
   } else if (io is OpenFile) {
-    return new File(io.path).open(mode: io.openForRead ? FileMode.READ : FileMode.WRITE).then((f) => new _RandomAccessFileRef(f));
+    return new File(io.path).open(mode: io.openForRead ? FileMode.read : FileMode.write).then((f) => new _RandomAccessFileRef(f));
```

Comment:
I know these cause deprecation warnings, but I've kept them so far since changing them breaks backward compatibility with Dart 1.

Reply:
Sorry, you mentioned before that you'd like to stay backwards compatible, but I totally forgot about that.